The Mac Weekly/utils.swift

```diff
@@ -138,6 +142,81 @@ public class TestableUtils {
         }
         
     }
+    
+    // Verify that a user's email meets our standards (ie is a mac.edu email)
+    static func validateLogin(email: String) -> Bool {
+        
+        let pattern = "^[A-Z0-9._%+-]+@macalester.edu$"
+        // The "^" and "$" characters indicate the start and end of the string, ensuring that the whole string must fit this pattern
+        // ALSO: two slashes are used here instead of one because swift gets weird about regular expressions
+        
+        let doesMatch = email.range(of: pattern, options: [.regularExpression, .caseInsensitive]) != nil
+        
+        return doesMatch
+        
+    }
+    
+    // Log in a user through Firebase
+    static func login(email: String, password: String) {
+        
+        print("About to sign in user")
+        Auth.auth().signIn(withEmail: email, password: password) { (user, error) in
+            
+            if let error = error {
+                print(error)
```

Comment:
Flagging this - Probably not something to solve for this merge, but at some point I think we want proper error handling. Going to add to trello